src/Select/SelectHelper.php

```diff
@@ -0,0 +1,99 @@
+
+        $model = new $modelName;
+
+        $selectColumns = [];
+
+        foreach ($fieldSelection as $field) {
+            $fieldDefinition = ASTHelper::firstByName($fieldDefinitions, $field);
+
+            if ($fieldDefinition) {
+                $directivesRequiringLocalKey = ['hasOne', 'hasMany', 'count'];
+                $directivesRequiringForeignKey = ['belongsTo', 'belongsToMany', 'morphTo'];
+                $directivesRequiringKeys = array_merge($directivesRequiringLocalKey, $directivesRequiringForeignKey);
+
+                foreach ($directivesRequiringKeys as $directiveType) {
+                    if (ASTHelper::hasDirective($fieldDefinition, $directiveType)) {
+                        $directive = ASTHelper::directiveDefinition($fieldDefinition, $directiveType);
+
+                        if (in_array($directiveType, $directivesRequiringLocalKey)) {
+                            $relationName = ASTHelper::directiveArgValue($directive, 'relation', $field);
+
+                            if (method_exists($model, $relationName)) {
+                                array_push($selectColumns, $model->{$relationName}()->getLocalKeyName());
+                            }
+                        }
+
+                        if (in_array($directiveType, $directivesRequiringForeignKey)) {
+                            $relationName = ASTHelper::directiveArgValue($directive, 'relation', $field);
+
+                            if (method_exists($model, $relationName)) {
+                                array_push($selectColumns, $model->{$relationName}()->getForeignKeyName());
+                            }
+                        }
+
+                        continue 2;
+                    }
+                }
+
+                if (ASTHelper::hasDirective($fieldDefinition, 'select')) {
+                    // append selected columns in select directive to seletion
+                    $directive = ASTHelper::directiveDefinition($fieldDefinition, 'select');
+
+                    if ($directive) {
+                        $selectFields = ASTHelper::directiveArgValue($directive, 'columns') ?? [];
+                        $selectColumns = array_merge($selectColumns, $selectFields);
+                    }
+                } elseif (ASTHelper::hasDirective($fieldDefinition, 'rename')) {
+                    // append renamed attribute to selection
+                    $directive = ASTHelper::directiveDefinition($fieldDefinition, 'rename');
+
+                    if ($directive) {
+                        $renamedAttribute = ASTHelper::directiveArgValue($directive, 'attribute');
+                        array_push($selectColumns, $renamedAttribute);
+                    }
+                } else {
+                    // fallback to selecting the field name
+                    array_push($selectColumns, $field);
+                }
+            }
+        }
+
+        return array_unique($selectColumns);
```

Comment:
I'm not sure, but if I need an accessor that is not in the table columns, we will get a "Column not found" SQL error.

```suggestion
        $allColumns = Schema::getColumnListing($model->getTable());
        return array_intersect($allColumns, array_unique($selectColumns));
```